DOCUMENTATION.md

```diff
@@ -1,157 +1,295 @@
-Most likely you don't want to show the data to your users in the product's
-main language. Most likely you want to prefer data in some specific
-language.
-
-Let's say you want a product to be displayed in German and
-so you've set the value of [AbstractQueryConfiguration.language] to German.
-Here's how the described above types of fields work in combinations:
-
-1. 1-1 (packaging info + simple value).
-   Fields like [Product.productName] will be in German if the German
-   versions of them are available in OFF.
-   If the German version is not available, the fields will be in the main
-   product's language.
-2. 1-2 (packaging info + multilingual value).
-   Fields like [Product.productNameInLanguages] will have the German value
-   inside if it's available.
-   If it's not available, `product.productNameInLanguages[GERMAN]`
-   will be `null`.
-3. 2-1 (meta info + simple value).
-   Fields like [Product.countriesTags] __will not be in German__, ever.
-   They will have values like `["en:austria", "en:belgium", "en:canada"]`.
-4. 2-2 (meta info + multilingual value).
-   Fields like [Product.countriesTagsInLanguages] will have German values
-   when available, if some of the values is not available in German it
-   will have a language prefix: `["Österreich ", "Belgien", "en:canada"]`.
-
-Those 4 combinations can be observed "in the wild" by
-executing next request:
-
-```url
-https://world.openfoodfacts.org/api/v2/product/3017620422003?lc=de&fields=product_name,product_name_de,countries_tags,countries_tags_de
-```
+-----
+
+## What is Open Food Facts?
+
+Open Food Facts is a collaborative, open-source project that collects data on millions of products worldwide. Contributors add and update information such as:
+
+  * **Ingredients**
+  * **Additives**
+  * **Brands**
+  * **Nutrition facts**
+  * **Serving sizes**
+  * **Nutri-Score** (for food products)
+  * **NOVA group** (for food products, concerning ultra-processing)
+  * **Green-Score** (for food products, regarding environmental impact)
+  * **Packaging details**
+  * **Certifications and labels**
+
+The database is constantly growing thanks to contributions from users like you, and all data is available under an open license.
+
+-----
+
+## How it Works
+
+The SDK leverages the Open (Food/PetFood/Beauty/Products) Facts APIs, which return product data in JSON format. These JSON responses are then converted into easy-to-understand Dart objects, simplifying their use in your applications. The SDK also enables you to edit existing products or upload new ones to the databases using the same straightforward product structures.
+
+-----
+
+## Understanding Language Mechanics
+
+When displaying or updating product data in specific languages, it's crucial to understand how the Open\*Facts APIs handle multilingual information. It is **highly recommended** to use the `..InLanguages` fields for both displaying and updating data to prevent accidental overwrites of existing language-specific data.
```

Comment:
```suggestion
When displaying or updating product data in specific languages, it's crucial to understand how the Open(Food/PetFood/Beauty/Products)Facts APIs handle multilingual information. It is **highly recommended** to use the `..InLanguages` fields for both displaying and updating data to prevent accidental overwrites of existing language-specific data.
```